confluence/client.py

```diff
@@ -477,6 +484,45 @@ def add_attachment(self, content_id, file_path, file_name=None, status=None):
             succeeded or not.
         """
         params = {}
+
+        if status:
+            if status in (ContentStatus.HISTORICAL, ContentStatus.TRASHED):
+                raise ValueError('Only draft or current are valid states for a new attachment')
+            params['status'] = status.value
+
+        if not file_name:
+            file_name = os.path.basename(file_path)
+
+        uri = 'content/{}/child/attachment'.format(content_id)
+
+        with open(file_path, 'rb') as f:
+            file = {
+                'file': (file_name, f),
+                'comment': "Uploaded automatically.",
+                'minorEdit': "true"
+            }
+
+            outputs = self._post_return_multiple(Content, uri, params=params, files=file, data={})
+
+        return outputs
+
+    def update_attachment(self, content_id, file_path, file_name=None, status=None):
```

Comment:
You need to make the comment and minorEdit optional parameters on this function signature rather than defaulting them below.

Reply:
Thanks! I'll get back to it soon. Your comments make a lot of sense though. Cheers!